row_073_file_route/row_073_file_route_safe.py

```
@@ -0,0 +1,41 @@
+@router.post("/upload-file-by-uploadfile",name="uploadfile")
+# {fact rule=unrestricted-file-upload@v1.0 defects=0}
+def uploadFile(file:UploadFile):
+    try:
+        UPLOAD_DIRECTORY = "./uploads/" # Ensure the directory exists 
+        os.makedirs(UPLOAD_DIRECTORY, exist_ok=True)
+        file_location = os.path.join(UPLOAD_DIRECTORY, file.filename)
+# defect
+        with open(file_location, "wb+") as file_object:
+            file_object.write(file.file.read())
+        return {"info": f"file '{file.filename}' saved at '{file_location}'"}
+    except Exception as e:
```

Ruff (0.14.2)

Do not catch blind exception: `Exception`

(BLE001)